pycbc/io/live.py

```diff
@@ -558,4 +564,321 @@ def gracedb_tag_with_version(gracedb, event_id):
+        "--duration-bin-end", type=float,
+        help="Longest duration to use for duration bins."
+    )
+    durbin_group.add_argument(
+        "--duration-from-bank",
+        help="Path to the template bank file to get max/min "
+             "durations from."
+    )
+    durbin_group.add_argument(
+        "--num-duration-bins",
+        type=int,
+        help="How many template duration bins to split the bank "
+             "into before fitting."
+    )
+    durbin_group.add_argument(
+        "--duration-bin-spacing",
+        choices=['linear', 'log'],
+        default='log',
+        help="How to set spacing for bank split "
+             "if using --num-duration-bins and "
+             "--duration-bin-start + --duration-bin-end "
+             "or --duration-from-bank."
+    )
+
+
+def verify_live_significance_duration_bin_options(args, parser):
+    """
+    Verify options used to calculate duration bin edges in live
+    singles significance fits
+    """
+    # Check the bin options
+    if args.duration_bin_edges:
+        if (args.duration_bin_start or args.duration_bin_end or
+                args.duration_from_bank or args.num_duration_bins):
+            parser.error("Cannot use --duration-bin-edges with "
+                         "--duration-bin-start, --duration-bin-end, "
+                         "--duration-from-bank or --num-duration-bins.")
+    else:
+        if not args.num_duration_bins:
+            parser.error("--num-duration-bins must be set if not using "
+                         "--duration-bin-edges.")
+        if not ((args.duration_bin_start and args.duration_bin_end) or
+                args.duration_from_bank):
+            parser.error("--duration-bin-start & --duration-bin-end or "
+                         "--duration-from-bank must be set if not using "
+                         "--duration-bin-edges.")
+    if args.duration_bin_end and \
+            args.duration_bin_end <= args.duration_bin_start:
+        parser.error("--duration-bin-end must be greater than "
+                     "--duration-bin-start, got "
+                     f"{args.duration_bin_end} and {args.duration_bin_start}")
+
+
+def find_trigger_files(directory, gps_start_time, gps_end_time,
+                       id_string='*', date_directories=False,
+                       date_directory_format="%Y_%m_%d"):
+    """
+    Find a list of PyCBC live trigger files which are between the gps
+    start and end times given
+    """
+
+    # Find the string at the start of the gps time which will match all
+    # files in this range - this helps to cut which ones we need to
+    # compare later
+    num_match = maximum_string([gps_start_time, gps_end_time])
+
+    # ** means recursive, so for large directories, this is expensive.
+    # It is not too bad if date_directories is set, as we don't waste time
+    # in directories where there cant be any files.
+    glob_string = f'**/*{id_string}*{num_match}*.hdf'
```

Comment:
I might be missing the point here, but I do not think it is worth the complication of trying to be smart with large directories. Storing more than a few thousand files in a single directory is terrible in terms of user experience, so it should never be done.

Reply:
I was thinking specifically of the early-warning trigger files, which have 86400 per directory - 10800 in the fullband ones is still a lot!

Reply:
Still not entirely convinced about the utility-complexity tradeoff, but fair enough.